### DISCUSSION TITLE
Add information about overtime

### DISCUSSION THREAD

**Discussion by wAuner** [jotaen/klog#385] [Feature ideas & discussions]
I noticed something that I'm really missing:
The sum of my overtime -not including- the current time. For example for `klog now -d`
```
            Today    Overall
Total          2m    165h27m
Should        8h!      168h!
Diff       -7h58m     -2h33m
E.T.A.      1:31>      20:06
```
I wish there was maybe a first line which would tell me what's the total of my overtime without taking the current time into account (or assuming I fulfill my should time today). Not sure how to best name and display it, but I think that's very important.

**Comment by wAuner** [comment]
the same also applies to `klog report -d`

**Comment by wAuner** [comment]
Maybe something along those lines
```
Time balance as of yesterday: +8h35m

            Today    Overall
Total          2m    165h27m
Should        8h!      168h!
Diff       -7h58m     -2h33m
E.T.A.      1:31>      20:06
```

**Comment by jotaen** [comment]
Just for clarification, so I can understand your use-case better: do you mean the diff without the current **_time_** or without the current **_day_**? Because in the original post you mention “without taking the current time into account”, but in the example you say “Time balance as of yesterday”.

So you are basically interested to also see the information that `klog total --diff --before 2021-03-07` would give you, is that right? (Assuming that `2021-03-07` yesterday’s date.)

**Comment by wAuner** [comment]
I guess it's more like the case without the current day, because I don't want to see today's "should time" included either.
Most time keeping accounts in companies show you your time balance where the current workday is not included. So that you always have an overview over how much overtime you have. This helps a lot for planning your workload/worktime and to me is absolute essential.

**Comment by wAuner** [comment]
On another note: `klog total --diff --before 2021-03-07` the docs say that the date specified by `--before` is inclusive. I find that very counterintuitive. If I say before today, then I'm almost always talking about the point up until today, not including today.

**Comment by jotaen** [comment]
> I guess it's more like the case without the current day, because I don't want to see today's "should time" included either.

Okay, thanks for clarifying. I think this makes sense, I’ll think about it!

> I find that very counterintuitive. If I say before today, then I'm almost always talking about the point up until today, not including today.

That’s a very fair point, I tend to agree 😆 Just out of curiosity: would you leave `--after` as is or change that as well? (It’s currently inclusive.) **UPDATE:** See https://github.com/jotaen/klog/issues/31 for further discussion.

**Comment by wAuner** [comment]
I‘d vote for after and before to be both exclusive

**Comment by jotaen** [comment]
I finally got around to thinking about this some more. (Actually, I also found myself needing this functionality now 😄 )

The `klog now` command treats the current day (either “today” or “yesterday”) as special and splits it off from the overall time. The missing link between “current” and “overall” is everything that happend before the current day. This could be reflected in a separate column called e.g. “Past”, which contains all previous records up until the current day. So, in mathematical terms, `Current + Past = Overall`

```
          Current       Past     Overall
          (Today)        
Total          2m    165h25m     165h27m
Should        8h!      160h!       168h!
Diff       -7h58m     +5h25m      -2h33m
E.T.A.      1:31>                  20:06

```

E.T.A. obviously doesn’t make much sense for the past.

**Comment by jotaen** [comment]
Also, I’m thinking about flipping the axes of the output table, because I think that would clarify the information design. So something along the lines of:

```
            Total    Should     Diff    End Time
Current        2m       8h!   -7h58m       1:31>
Past      165h25m     160h!   +5h25m
         ========  ======== ========
          165h27m     168h!   -2h33m       20:06
```

That way it’s more obvious that the last line is basically the sum of the previous ones.

**Comment by wAuner** [comment]
I like the flipped axes version, it's more intuitive.
